### PR TITLE
chore: Add unit test for state-sriov-dp

### PR DESCRIPTION
**Pull request by souleb**

### CODE REVIEW

**Comment by rollandf** [comment]
Can you please check the `lint` error?

**Review by rollandf — DISMISSED** [review]

File: pkg/state/state_sriov_dp_test.go [hunk 1]
```diff
@@ -14,130 +14,280 @@ See the License for the specific language governing permissions and
...
+			err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace,
+				Name: "sriov-device-plugin"}, ds)
+			Expect(err).NotTo(HaveOccurred())
+			assertCommonDaemonSetFields(ds, &cr.Spec.SriovDevicePlugin.ImageSpec, cr)
+			// expect privileged mode
+			Expect(*ds.Spec.Template.Spec.Containers[0].SecurityContext.Privileged).To(BeTrue())
+			assertSriovDpPodTemplatesVolumeFields(&ds.Spec.Template, false)
+			assertSriovDpPodTemplatesVolumeMountFields(&ds.Spec.Template, false)
+		})
+		It("should create Daemonset with CDI support when specified in CR", func() {
```
Comment (adrianchiris) [inline pkg/state/state_sriov_dp_test.go:82]: can you add a check that device plugin is started with "--use-cdi" flag ? unless i missed it in the test :)

File: pkg/state/state_sriov_dp_test.go [hunk 2]
```diff
@@ -14,130 +14,280 @@ See the License for the specific language governing permissions and
...
+			cr := getMinimalNicClusterPolicyWithSriovDp(false)
+			status, err := sriovDpState.Sync(context.Background(), cr, catalog)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(status).To(BeEquivalentTo(state.SyncStateNotReady))
+			By("Verify DaemonSet")
+			ds := &appsv1.DaemonSet{}
+			err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace,
+				Name: "sriov-device-plugin"}, ds)
+			Expect(err).NotTo(HaveOccurred())
+			assertCommonDaemonSetFields(ds, &cr.Spec.SriovDevicePlugin.ImageSpec, cr)
```
Comment (adrianchiris) [inline pkg/state/state_sriov_dp_test.go:112]: nit: we dont really need to re check what was checked in previous cases. just the status here imo.
that said. up to u.

File: pkg/state/state_sriov_dp_test.go [hunk 3]
```diff
@@ -14,130 +14,280 @@ See the License for the specific language governing permissions and
...
+			status, err := sriovDpState.Sync(context.Background(), cr, catalog)
 			Expect(err).NotTo(HaveOccurred())
-			Expect(sriovDpState.Name()).To(Equal(stateName))
+			Expect(status).To(BeEquivalentTo(state.SyncStateNotReady))
+			By("Verify DaemonSet")
+			ds := &appsv1.DaemonSet{}
+			err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace,
+				Name: "sriov-device-plugin"}, ds)
+			Expect(err).NotTo(HaveOccurred())
+			assertCommonDaemonSetFields(ds, &cr.Spec.SriovDevicePlugin.ImageSpec, cr)
```
Comment (adrianchiris) [inline pkg/state/state_sriov_dp_test.go:143]: same as above wrt checking volumes, common fields and mounts...

**Review by adrianchiris — COMMENTED** [review]
nice work @souleb ! added a couple of small comments

File: pkg/state/state_sriov_dp_test.go (hunk 1 above)
Comment (souleb) [reply pkg/state/state_sriov_dp_test.go:82]: That is what this test is about. `cr := getMinimalNicClusterPolicyWithSriovDp(true)` sets `useCDI = true` in the devicePlugin spec.

Comment (adrianchiris) [reply pkg/state/state_sriov_dp_test.go:82]: please show me the line where the test check that "--use-cdi" flag is passed to device plugin command in the daemonset.

Comment (souleb) [reply pkg/state/state_sriov_dp_test.go:82]: Ok, I just added it.

**Review by adrianchiris — APPROVED** [review]

**Comment by rollandf** [comment]
`nic_operator_kind CI` fails on  `"test-nv-peer-mem-with-host-device": "Failed"`
It is failing for the same in other PRs.

**Pull request merged**